core/types/transaction_signing.go

```diff
@@ -38,12 +38,15 @@ type sigCache struct {
 
 // MakeSigner returns a Signer based on the given chain config and block number.
 func MakeSigner(config *params.ChainConfig, blockNumber *big.Int, blockTime uint64) Signer {
+
+	if signer, isCelo := makeCeloSigner(config, blockNumber, blockTime); isCelo {
```

Comment:
Sounds good to me!

Reply:
Although maybe Hook is not the best postfix, because I see a lot of xxxHook things already in the codbase, it would be nice to use something distinct to aid finding all the hookpoints.

Reply:
yeah, don't mind the name... more on something that doesn't say celo, because it's the extension point to any other op chain.